### ISSUE TITLE
DNS and Storage addons not ready but ready

### ISSUE THREAD

**Issue by Moeletji17** [ubuntu/microk8s#897]
I have been struggling with a fresh installation of microk8s.

The command used for installation:
snap install microk8s --classic

I enabled the dns and storage addons using:
microk8s.enable dns storage

I get the following when I check all the namespaces (kubectl get all --all-namespaces):

NAMESPACE     NAME                                        READY   STATUS    RESTARTS   AGE
kube-system   pod/coredns-9b8997588-2nc67                 0/1     Running   0          3m5s
kube-system   pod/hostpath-provisioner-7b9cb5cdb4-c6bgq   1/1     Running   3          2m58s

 iptables is used as the firewall and ufw is disabled.

Please assist with resolving this issue as the link: https://microk8s.io/docs/troubleshooting#common-issues doesn't provide any solutions to the problem.
[inspection-report-20200107_163326.tar.gz](https://github.com/ubuntu/microk8s/files/4035188/inspection-report-20200107_163326.tar.gz)

**Comment by ktsakalozos** [comment]
Hi @Moeletji17 could you create a new `microk8s.inspect` tarball and attach it to the issue? The one already attached is with the cluster empty. Thanks.

**Comment by Moeletji17** [comment]
Apologies for the oversight. I had been reinstalling it multiple times trying out multiple releases.

Please find the updated inspection file attached. 
[inspection-report-20200108_143527.tar.gz](https://github.com/ubuntu/microk8s/files/4035305/inspection-report-20200108_143527.tar.gz)

**Comment by ktsakalozos** [comment]
I see in the logs of the coredns pod:
```
E0108 12:35:11.773577       1 reflector.go:134] pkg/mod/k8s.io/client-go@v10.0.0+incompatible/tools/cache/reflector.go:95: Failed to list *v1.Endpoints: Get https://10.152.183.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.152.183.1:443: i/o timeout
```
This means the pod cannot reach the api server. Can you share the output of `sudo iptables -S` and/or `sudo iptables -L`. 

I see also you are following the latest stable track but your cluster is not an ephemeral one. The latest stable track may be introducing backwards incompatible changes overtime. For non-ephemeral clusters you could consider following a specific track with for example `sudo snap install microk8s --classic --channel=1.17/stable`.

**Comment by Moeletji17** [comment]
The X.X.X.X is the host.

I had the same issue with the stable release but let me try it again.

[iptables-latest.txt](https://github.com/ubuntu/microk8s/files/4035563/iptables-latest.txt)

**Comment by Moeletji17** [comment]
> I see in the logs of the coredns pod:
> 
> ```
> E0108 12:35:11.773577       1 reflector.go:134] pkg/mod/k8s.io/client-go@v10.0.0+incompatible/tools/cache/reflector.go:95: Failed to list *v1.Endpoints: Get https://10.152.183.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.152.183.1:443: i/o timeout
> ```
> 
> This means the pod cannot reach the api server. Can you share the output of `sudo iptables -S` and/or `sudo iptables -L`.
> 
> I see also you are following the latest stable track but your cluster is not an ephemeral one. The latest stable track may be introducing backwards incompatible changes overtime. For non-ephemeral clusters you could consider following a specific track with for example `sudo snap install microk8s --classic --channel=1.17/stable`.

I tried following the installation for non-ephemeral clusters as suggested using v1.17 and I still had the same issue

**Comment by Moeletji17** [comment]
> I see in the logs of the coredns pod:
> 
> ```
> E0108 12:35:11.773577       1 reflector.go:134] pkg/mod/k8s.io/client-go@v10.0.0+incompatible/tools/cache/reflector.go:95: Failed to list *v1.Endpoints: Get https://10.152.183.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.152.183.1:443: i/o timeout
> ```
> 
> This means the pod cannot reach the api server. Can you share the output of `sudo iptables -S` and/or `sudo iptables -L`.
> 
> I see also you are following the latest stable track but your cluster is not an ephemeral one. The latest stable track may be introducing backwards incompatible changes overtime. For non-ephemeral clusters you could consider following a specific track with for example `sudo snap install microk8s --classic --channel=1.17/stable`.

Would I need to add a rule to the iptables for the pods to connect to the apiserver?

**Comment by ktsakalozos** [comment]
The IPtables rules for pod connections are in place but their priority might need to be raised:
```
ACCEPT     all  --  10.1.0.0/16          anywhere             /* generated for MicroK8s pods */
ACCEPT     all  --  anywhere             10.1.0.0/16          /* generated for MicroK8s pods */
ACCEPT     all  --  10.1.0.0/16          anywhere
ACCEPT     all  --  anywhere             10.1.0.0/16
```

I am a bit skeptical about the DOCKER-ISOLATION chain. Can you try re-adding the 10.1.0.0/16 rules at the top of the forward chan? See https://www.cyberciti.biz/faq/linux-iptables-insert-rule-at-top-of-tables-prepend-rule/ for example.

**Comment by Moeletji17** [comment]
I've applied the suggested solutions to the cluster(with the 1.17/stable channel):

- Moving the IPtables rules for pod connections to the top of the FORWARD chain and then stop and start microk8s, but the result is still the same. 

- Additionally I removed the DOCKER ISOLATION chain to resemble a test environment where I configured Microk8s using ufw then disabled it and used iptables but the same result as the initial post. 

Is there a well documented reference for Microk8s networking so I can get a better understanding which may help me understand my problem better? 

I've been struggling with this issue for a while. It seems the best way for configuring Microk8s is using ufw because
